pkg/skaffold/errors/err_map.go

```diff
@@ -87,3 +88,26 @@ var knownBuildProblems = []problem{
 		},
 	},
 }
+
+// Deploy errors in deployment phase
+var knownDeployProblems = []problem{
+	{
+		regexp:  re("(?i).*unable to connect.*: Get (.*)"),
+		errCode: proto.StatusCode_DEPLOY_CLUSTER_CONNECTION_ERR,
+		description: func(err error) string {
+			matchExp := re("(?i).*unable to connect.*Get (.*)")
+			kubeconfig, _ := kubectx.CurrentConfig()
+			if match := matchExp.FindStringSubmatch(fmt.Sprintf("%s", err)); len(match) >= 2 {
+				// kubeconfig, _ := kubectx.CurrentConfig()
+				return fmt.Sprintf("Deploy Failed. Could not connect to cluster %s due to %s", kubeconfig.CurrentContext, match[1])
+			}
+			return fmt.Sprintf("Deploy Failed. Could not connect to %s cluster.", kubeconfig.CurrentContext)
+		},
+		suggestion: func(opts config.SkaffoldOptions) []*proto.Suggestion {
+			return []*proto.Suggestion{{
+				SuggestionCode: proto.SuggestionCode_CHECK_HOST_CONNECTION,
```

Comment:
Can we use the same function `kubectx.CurrentConfig()` to determine the active context and suggest different actions when context is minikube vs GKE.
you can use the `cluster.isMinikube` method to determine if this is a minikube cluster

Reply:
Updated.